### PR TITLE
Document the sum_match argument of jax.vmap

### PR DESCRIPTION
**Pull request by kodlan**
Part of #39159, following the maintainer guidance there: this documents only
sum_match (spmd_axis_name is left undocumented intentionally, pending
deprecation/redesign).
Adds an Args entry for sum_match with a doctest example: with
out_axes=None, outputs that vary along the mapped axis are summed over it
instead of raising an error (outputs that don't vary are returned unchanged);
useful in autodiff contexts because summing over a mapped axis is the
transpose of broadcasting along it. Also cross-references sum_match from
the out_axes paragraph whose "must have non-None out_axes" statement it
relaxes.

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Beautiful, thank you!

**Comment by mattjj** [comment]
Would you mind squashing the commits?

**Comment by kodlan** [comment]
> Would you mind squashing the commits?

Done. Sorry forgot that I've applied some suggestion from gemini-code-assist here..

**Comment by kodlan** [comment]
Hm something is failing for cl/950174218

**Comment by kodlan** [comment]
@mattjj could you please restart/ignore ROCm tests? Thanks!

**Comment by mattjj** [comment]
Those shouldn’t block (ie they’re ignored) and are failing already I think, but yes I’ll restart them.

**Comment by mattjj** [comment]
Actually let me just try to get this merged when I get to a keyboard…

**Comment by kodlan** [comment]
Thank you!
